### PR TITLE
Allow github host and prefix to be set via environment variable

### PR DESCRIPTION
**Pull request by skaasten**
Rather than edit the package.json config, it's nice to use environment variables, particularly when running in a docker container.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -50,8 +50,8 @@ if (!process.env.GITHUB_USER) {
 
 var github = new GitHubApi({
   version: '3.0.0',
-  host: config.gheHost,
-  pathPrefix: config.ghePathPrefix,
+  host: process.env.GITHUB_HOST || config.gheHost,
+  pathPrefix: process.env.GITHUB_API_PREFIX || config.ghePathPrefix,
```
Comment (vjeux) [inline server.js]: Can you respect the existing naming so that we don't have two different ones?

GHE_PATH_PREFIX
GHE_HOST

**Comment by vjeux** [comment]
This is a bit brittle to check this for all the call sites. What would be better is to make a file config.js that returns all the fields and uses the env variable if exists.

This way we don't forget to check the env variable the next time we use an attribute.

Also, would be great to have env variables for all the options, instead of a few arbitrary that you decided?

**Comment by skaasten** [comment]
Maybe it could use something like https://github.com/lorenwest/node-config? I can update the PR with it.

**Comment by vjeux** [comment]
Isn't there a way with docker to change the package.json values?

**Comment by skaasten** [comment]
I would prefer to not update package.json each time there's a new
mention-bot release - env variables (or env files) are the standard way to
pass in creds to a docker container app.

On Fri, Mar 4, 2016 at 9:59 AM, Christopher Chedeau <
notifications@github.com> wrote:

> Isn't there a way with docker to change the package.json values?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/mention-bot/pull/90#issuecomment-192311736>.
>



-- 
Shaun

**Comment by vjeux** [comment]
Okay makes sense. Can you add a way to load the data from a json file passed as argument instead of relying on the env variables?

```
npm run mention-bot --config ./config.json
```

this way we don't have to have 2 ways to specify configs, we can either use package.json (default) or config.json

**Comment by skaasten** [comment]
:+1:  - I'll work on that here: https://github.com/facebook/mention-bot/compare/master...freshbooks:environment-config?expand=1 and put up a new PR.

**Pull request closed without merging**